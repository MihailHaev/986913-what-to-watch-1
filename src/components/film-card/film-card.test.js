import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card";

const mock = {
  film: {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: ``,
  }
};

it(`Film Card correctly renders`, () => {
  const {film} = mock;
  const tree = renderer
    .create(<FilmCard
      film={film}
      checkActiveCard={jest.fn()}
      onMouseOut={jest.fn()}
      onMouseOver={jest.fn()}
      key={0}
      id={0}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
