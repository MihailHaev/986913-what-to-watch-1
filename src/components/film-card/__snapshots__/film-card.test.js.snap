// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Card correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <video
    height="175"
    muted={true}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    src=""
    width="280"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
